cou/utils/juju_utils.py

```diff
@@ -494,11 +480,21 @@ async def update_status(self, unit_name: str) -> None:
         :raises CommandRunFailed: When update-status hook failed
         """
         # For charm written in operator framework
-        if await self._dispatch_update_status_hook(unit_name):
+        try:
+            await self._dispatch_update_status_hook(unit_name)
+        except CommandRunFailed as e:
+            if "No such file or directory" not in str(e):
+                raise e
+        else:
             return
```

Comment:
nonblocking nit: the `return` could be moved to directly underneath `await self._dispatch_update_status_hook(unit_name)`, removing the need for the `else:`.  same for the next try/except/else block.

Reply:
yeah, I don't really think that will help gaining much performance here or help understanding the code better, and it looks cleaner to me that we want to exit if `try` block succeeded. I would like to keep it this way

Reply:
no worries